### PR TITLE
added liquibase support, implemented soft delete concept, added new endpoint

### PR DESCRIPTION
**Pull request by SvNadorozhnyi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +58,16 @@
             <artifactId>mapstruct</artifactId>
             <version>${mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>4.29.2</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
            <version>${liquibase.version}</version>
```

File: pom.xml [hunk 2]
```diff
@@ -58,6 +58,16 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-maven-plugin</artifactId>
+            <version>4.29.2</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
            <version>${liquibase.version}</version>
```

File: src/main/java/org/example/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -32,4 +36,15 @@ public BookDto getBookById(@PathVariable Long id) {
...
     }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteById(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable Long id, @RequestBody BookDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/controller/BookController.java]:
```suggestion
    public BookDto updateBookById(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```

File: src/main/java/org/example/bookstore/model/Book.java [hunk 4]
```diff
@@ -8,10 +8,14 @@
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
 
 @Entity
 @Data
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java]: Don't use @Data for entities

```suggestion
```

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -38,4 +38,20 @@ public BookDto findById(Long id) {
...
+
+    @Override
+    public BookDto updateBookById(Long id, BookDto bookDto) {
+        if (bookRepository.existsById(id)) {
+            bookDto.setId(id);
+            bookRepository.save(bookMapper.toModel(bookDto));
+            return bookDto;
+        } else {
+            throw new EntityNotFoundException("Can't find book by id " + id);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(()....
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:47]: add one empty line to the end of the file (check other files)

**Review by liliia-ponomarenko — APPROVED** [review]
Awesome 😎

**Pull request merged**